src/cmd/run.js

```diff
@@ -62,24 +65,36 @@ export function defaultWatcherCreator(
   return onSourceChange({
     sourceDir,
     artifactsDir,
-    onChange: () => {
-      log.debug(`Reloading add-on ID ${addonId}`);
-
-      return client.reloadAddon(addonId)
-        .catch((error) => {
-          log.error('\n');
-          log.error(error.stack);
-          desktopNotifications({
-            title: 'web-ext run: error occurred',
-            message: error.message,
-          });
-          throw error;
-        });
-    },
+    onChange: () => addonReload({addonId, client}),
```

Comment:
Actually, I was wrong, I was forgetting that the `addReload` utility function is needed by the keypress handler.

@saintsebastian As per discussion over email, you can ignore this comment and only handle the rest of the review comments.